qiskit/circuit/library/graph_state.py

```diff
@@ -84,3 +90,87 @@ def __init__(self, adjacency_matrix: list | np.ndarray) -> None:
 
         super().__init__(*circuit.qregs, name=circuit.name)
         self.compose(circuit.to_gate(), qubits=self.qubits, inplace=True)
+
+
+class GraphStateGate(Gate):
+    r"""A gate representing a graph state.
+
+    Given a graph G = (V, E), with the set of vertices V and the set of edges E,
+    the corresponding graph state is defined as
+
+    .. math::
+
+        |G\rangle = \prod_{(a,b) \in E} CZ_{(a,b)} {|+\rangle}^{\otimes V}
+
+    Such a state can be prepared by first preparing all qubits in the :math:`+`
+    state, then applying a :math:`CZ` gate for each corresponding graph edge.
+
+    Graph state preparation circuits are Clifford circuits, and thus
+    easy to simulate classically. However, by adding a layer of measurements
+    in a product basis at the end, there is evidence that the circuit becomes
+    hard to simulate [2].
+
+    **Reference Circuit:**
+
+    .. plot::
+
```

Comment:
This applies to all other docstrings as well: What do you think about including the source to show how this object can be used? (I think you need to re-indent the code with this flag, so better not accept this suggestion)
```suggestion
    .. plot::
        :include-source:
```

Reply:
Done here and in all other files.